From: Jacob Keller <jacob.e.keller@intel.com>
Date: Fri, 26 Oct 2018 14:33:29 -0700
Subject: i40e: always set ks->base.speed in i40e_get_settings_link_up
Patch-mainline: v5.0-rc1
Git-commit: 61bfb0600551808ff12a1e06795ef960f0544a36
References: bsc#1111981 FATE#326312 FATE#326313

In i40e_get_settings_link_up, set ks->base.speed to SPEED_UNKNOWN
in the case where we don't know the link speed.

Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Tested-by: Andrew Bowers <andrewx.bowers@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/intel/i40e/i40e_ethtool.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -929,6 +929,7 @@ static void i40e_get_settings_link_up(st
 		ks->base.speed = SPEED_100;
 		break;
 	default:
+		ks->base.speed = SPEED_UNKNOWN;
 		break;
 	}
 	ks->base.duplex = DUPLEX_FULL;
